src/formatters/stylishFormatter.ts

```diff
@@ -26,15 +26,24 @@ export class Formatter extends AbstractFormatter {
             return "\n";
         }
 
-        const fileName        = failures[0].getFileName();
+        const outputLines: Array<string> = [];
+
         const positionMaxSize = this.getPositionMaxSize(failures);
         const ruleMaxSize     = this.getRuleMaxSize(failures);
 
-        const outputLines = [
-            fileName,
-        ];
+        let currentFile: string;
 
         for (const failure of failures) {
+            // File name for each new file
+            const fileName = failure.getFileName();
+            if (currentFile !== fileName) {
+                if (currentFile) {
+                    outputLines.push("");
```

Comment:
It happens every time a new file is reached.
The idea is that whenever we start a new file, we add an empty line before it to separate the output from the previous list of lint errors.
I'll refactor it out of the loop by adding the first file-name before it.